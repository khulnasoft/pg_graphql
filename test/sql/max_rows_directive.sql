begin;
    create table account(
        id int primary key
    );

    insert into public.account(id)
    select * from generate_series(1, 100);

    -- expect default 30 rows on first page
    select graphql.resolve($$
      {
        accountCollection
        {
          edges {
            node {
              id
            }
          }
        }
      }
    $$);

    comment on schema public is e'@graphql({"max_rows": 5})';

    -- expect 5 rows on first page
    select graphql.resolve($$
      {
        accountCollection
        {
          edges {
            node {
              id
            }
          }
        }
      }
    $$);

    comment on schema public is e'@graphql({"max_rows": 40})';

    -- expect 40 rows on first page
    select graphql.resolve($$
      {
        accountCollection
        {
          edges {
            node {
              id
            }
          }
        }
      }
    $$);

rollback;
