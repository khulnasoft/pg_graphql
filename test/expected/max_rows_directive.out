begin;
    create table account(
        id int primary key
    );
    insert into public.account(id)
    select * from generate_series(1, 100);
    -- expect default 30 rows on first page
    select graphql.resolve($$
      {
        accountCollection
        {
          edges {
            node {
              id
            }
          }
        }
      }
    $$);
                                                                                                                                                                                                                                                                                                                                                         resolve                                                                                                                                                                                                                                                                                                                                                         
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {"data": {"accountCollection": {"edges": [{"node": {"id": 1}}, {"node": {"id": 2}}, {"node": {"id": 3}}, {"node": {"id": 4}}, {"node": {"id": 5}}, {"node": {"id": 6}}, {"node": {"id": 7}}, {"node": {"id": 8}}, {"node": {"id": 9}}, {"node": {"id": 10}}, {"node": {"id": 11}}, {"node": {"id": 12}}, {"node": {"id": 13}}, {"node": {"id": 14}}, {"node": {"id": 15}}, {"node": {"id": 16}}, {"node": {"id": 17}}, {"node": {"id": 18}}, {"node": {"id": 19}}, {"node": {"id": 20}}, {"node": {"id": 21}}, {"node": {"id": 22}}, {"node": {"id": 23}}, {"node": {"id": 24}}, {"node": {"id": 25}}, {"node": {"id": 26}}, {"node": {"id": 27}}, {"node": {"id": 28}}, {"node": {"id": 29}}, {"node": {"id": 30}}]}}}
(1 row)

    comment on schema public is e'@graphql({"max_rows": 5})';
    -- expect 5 rows on first page
    select graphql.resolve($$
      {
        accountCollection
        {
          edges {
            node {
              id
            }
          }
        }
      }
    $$);
                                                                        resolve                                                                        
-------------------------------------------------------------------------------------------------------------------------------------------------------
 {"data": {"accountCollection": {"edges": [{"node": {"id": 1}}, {"node": {"id": 2}}, {"node": {"id": 3}}, {"node": {"id": 4}}, {"node": {"id": 5}}]}}}
(1 row)

    comment on schema public is e'@graphql({"max_rows": 40})';
    -- expect 40 rows on first page
    select graphql.resolve($$
      {
        accountCollection
        {
          edges {
            node {
              id
            }
          }
        }
      }
    $$);
                                                                                                                                                                                                                                                                                                                                                                                                                                                                       resolve                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {"data": {"accountCollection": {"edges": [{"node": {"id": 1}}, {"node": {"id": 2}}, {"node": {"id": 3}}, {"node": {"id": 4}}, {"node": {"id": 5}}, {"node": {"id": 6}}, {"node": {"id": 7}}, {"node": {"id": 8}}, {"node": {"id": 9}}, {"node": {"id": 10}}, {"node": {"id": 11}}, {"node": {"id": 12}}, {"node": {"id": 13}}, {"node": {"id": 14}}, {"node": {"id": 15}}, {"node": {"id": 16}}, {"node": {"id": 17}}, {"node": {"id": 18}}, {"node": {"id": 19}}, {"node": {"id": 20}}, {"node": {"id": 21}}, {"node": {"id": 22}}, {"node": {"id": 23}}, {"node": {"id": 24}}, {"node": {"id": 25}}, {"node": {"id": 26}}, {"node": {"id": 27}}, {"node": {"id": 28}}, {"node": {"id": 29}}, {"node": {"id": 30}}, {"node": {"id": 31}}, {"node": {"id": 32}}, {"node": {"id": 33}}, {"node": {"id": 34}}, {"node": {"id": 35}}, {"node": {"id": 36}}, {"node": {"id": 37}}, {"node": {"id": 38}}, {"node": {"id": 39}}, {"node": {"id": 40}}]}}}
(1 row)

rollback;
